tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1204,6 +1232,156 @@ Status IrEmitterUnnested::EmitConvolutionReorderThunk(mlir::Operation* op) {
+      // N.B. tuning_knobs is a map rather than a repeated field, so this
+      // doesn't require reserving space up front.
+      (*algorithm->mutable_tuning_knobs())[fmha.getAlgorithmConfig()
+                                               .getKnobIds()[i]] =
+          fmha.getAlgorithmConfig().getKnobValues()[i];
+    }
+    algorithm->set_is_cudnn_frontend(true);
+    auto workspace_size = fmha.getAlgorithmConfig().getWorkspaceSize();
+    if (workspace_size >= 0) {
+      algorithm->mutable_workspace_size()->set_value(workspace_size);
+    }
+
+    descriptor.bmm1_dnums =
+        ConvertDotDimensionNumbers(fmha.getBmm1DotDimensionNumbers());
+    descriptor.bmm2_dnums =
+        ConvertDotDimensionNumbers(fmha.getBmm2DotDimensionNumbers());
+
+    descriptor.lhs_bmm1_shape = ShapeUtil::MakeShapeWithDenseLayout(
+        GetShape(fmha.getLhsBmm1()).element_type(),
+        GetShape(fmha.getLhsBmm1()).dimensions(),
+        GetShape(fmha.getLhsBmm1()).layout().minor_to_major());
+    TF_ASSIGN_OR_RETURN(lhs_bmm1_slice, GetAllocationSlice(fmha.getLhsBmm1()));
+
+    descriptor.rhs_bmm1_shape = ShapeUtil::MakeShapeWithDenseLayout(
+        GetShape(fmha.getRhsBmm1()).element_type(),
+        GetShape(fmha.getRhsBmm1()).dimensions(),
+        GetShape(fmha.getRhsBmm1()).layout().minor_to_major());
+    TF_ASSIGN_OR_RETURN(rhs_bmm1_slice, GetAllocationSlice(fmha.getRhsBmm1()));
+
+    descriptor.rhs_bmm2_shape = ShapeUtil::MakeShapeWithDenseLayout(
+        GetShape(fmha.getRhsBmm2()).element_type(),
+        GetShape(fmha.getRhsBmm2()).dimensions(),
+        GetShape(fmha.getRhsBmm2()).layout().minor_to_major());
+    TF_ASSIGN_OR_RETURN(rhs_bmm2_slice, GetAllocationSlice(fmha.getRhsBmm2()));
+
+    descriptor.output_shape = ShapeUtil::MakeShapeWithDenseLayout(
+        GetShape(fmha.getOutput()).element_type(),
+        GetShape(fmha.getOutput()).dimensions(),
+        GetShape(fmha.getOutput()).layout().minor_to_major());
+    TF_ASSIGN_OR_RETURN(output_slice, GetAllocationSlice(fmha.getOutput()));
+
+    TF_ASSIGN_OR_RETURN(scratch_slice, GetAllocationSlice(fmha.getScratch()));
+
+    TF_ASSIGN_OR_RETURN(auto intermediate_tensor_dims_array,
+                        ConvertMlirArrayAttrToInt64Array(
+                            fmha.getIntermediateTensorDimensions()));
+    TF_ASSIGN_OR_RETURN(
+        auto intermediate_tensor_layout_array,
+        ConvertMlirArrayAttrToInt64Array(fmha.getIntermediateTensorLayout()));
+
+    descriptor.intermediate_lhs_bmm2_shape =
+        ShapeUtil::MakeShapeWithDenseLayout(
+            GetShape(fmha.getOutput()).element_type(),
+            intermediate_tensor_dims_array, intermediate_tensor_layout_array);
+    return OkStatus();
+  };
+
+  enum OptionalBuffer {
+    kBias,
+    kMask,
+  };
+  BufferAllocation::Slice mask_slice;
+  BufferAllocation::Slice bias_slice;
+  auto allocate_buffer = [&](auto fmha, OptionalBuffer optional_buffer_kind) {
```

Comment:
The name seems misleading: the function is not allocating anything, right?

Reply:
Changed to `get_buffer_allocation_slice`. Done.